autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/imputation/base_imputer.py

```diff
@@ -26,7 +25,7 @@ def transform(self, X: Dict[str, Any]) -> Dict[str, Any]:
         Returns:
             (Dict[str, Any]): the updated 'X' dictionary
         """
-        if self.preprocessor['numerical'] is None and self.preprocessor['categorical'] is None:
+        if self.preprocessor['numerical'] is None and len(X["dataset_properties"]["numerical_columns"]) != 0:
             raise ValueError("cant call transform on {} without fitting first."
```

Comment:
```suggestion
            raise ValueError("cannot call transform on {} without fitting first."
```

Reply:
"can not" is two words :)

Reply:
I did not know that "can not" exists, but "cannot" is preferred for formal settings?

- [article1](https://www.grammarly.com/blog/cannot-or-can-not/) --> "You’ll find cannot in formal writing and speech. Research studies, academic reports, and professional presentations are places where cannot is the most appropriate choice."
- [article 2](https://grammar.yourdictionary.com/vs/cannot-vs-can-not-vs-cant-differences-explained.html) --> "While can not is an acceptable alternate spelling, cannot is generally preferred by most writers."
- [article 3](https://www.merriam-webster.com/words-at-play/cannot-vs-can-not-is-there-a-difference) --> "Both cannot and can not are perfectly fine, but cannot is far more common and is therefore recommended"